influxql/parser.go

```diff
@@ -2536,41 +2536,66 @@ func ParseDuration(s string) (time.Duration, error) {
 	// Split string into individual runes.
 	a := split(s)
 
-	// Extract the unit of measure.
-	// If the last two characters are "ms" then parse as milliseconds.
-	// Otherwise just use the last character as the unit of measure.
-	var num, uom string
-	if len(s) > 2 && s[len(s)-2:] == "ms" {
-		num, uom = string(a[:len(a)-2]), "ms"
-	} else {
-		num, uom = string(a[:len(a)-1]), string(a[len(a)-1:])
-	}
+	// Start with a zero duration.
```

Comment:
I know it's not part of your change-set, but could you add detail to the function's doc-string?  I was staring at this function wondering why it was even needed over `time.ParseDuration`, and it took me a while to figure out that it's only because we support days and weeks.

I think it's worth explaining that in the function comment.